modules/xfem/src/userobjects/GeometricCut2DUserObject.C

```diff
@@ -0,0 +1,227 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "GeometricCut2DUserObject.h"
+
+// MOOSE includes
+#include "MooseError.h"
+
+// libMesh includes
+#include "libmesh/string_to_enum.h"
+
+// XFEM includes
+#include "XFEMFuncs.h"
+
+template <>
+InputParameters
+validParams<GeometricCut2DUserObject>()
+{
+  // Get input parameters from parent class
+  InputParameters params = validParams<GeometricCutUserObject>();
+
+  // Class description
+  params.addClassDescription("Base class for 2D XFEM Geometric Cut UserObjects");
+  // Return the parameters
+  return params;
+}
+
+GeometricCut2DUserObject::GeometricCut2DUserObject(const InputParameters & parameters)
+  : GeometricCutUserObject(parameters), _cut_line_start_points(), _cut_line_end_points()
+{
+}
+
+GeometricCut2DUserObject::~GeometricCut2DUserObject() {}
+
+bool
+GeometricCut2DUserObject::active(Real time) const
+{
+  bool is_cut_active = false;
+
+  for (unsigned int cut = 0; cut < _cut_line_start_points.size(); ++cut)
+    if (cutFraction(cut, time) > 0)
+      is_cut_active = true;
+
+  return is_cut_active;
+}
+
+bool
+GeometricCut2DUserObject::cutElementByGeometry(const Elem * elem,
+                                               std::vector<CutEdge> & cut_edges,
+                                               Real time) const
+{
+  bool cut_elem = false;
+
+  out << "\n";
+  out << "geom_cut_2d _start_times: ";
+  for (unsigned int i = 0; i < _start_times.size(); ++i)
+  {
+    out << _start_times[i];
+    out << ", ";
```

Comment:
This is OK, but you can consolidate this into a single line rather elegantly with the right commas. 
```std::copy(_start_times.begin(), _start_times.end(), infix_ostream_iterator<std::string>(oss, ", "));```
or even better
https://github.com/idaholab/moose/blob/devel/framework/include/utils/Conversion.h#L87-L97